src/components/Home/Today1Hourly/Today1Hourly.tsx

```diff
@@ -0,0 +1,142 @@
+  tempUnit: TempUnits;
+}
+interface Today1HourlyState {
+  data: HourlyData;
+  loading: boolean;
+  error: string;
+}
+
+class Today1Hourly extends React.Component<Today1HourlyProps, Today1HourlyState> {
+  constructor(props: Today1HourlyProps) {
+    super(props);
+    this.state = {
+      data: { forecast: [] },
+      loading: false,
+      error: ''
+    };
+  }
+
+  getHourlyData(): void {
+    getData(ENDPOINTS.hourly, this.props.location).then(
+      (data: HourlyData) => {
+        this.setState(() => ({ loading: true, data }));
+      },
+      (error: Error | AxiosError) => {
+        this.setState(() => ({ error: error.message }));
+      }
+    );
+  }
+
+  componentDidMount() {
+    this.getHourlyData();
+  }
+  componentDidUpdate(prevProps: { location: string }) {
+    if (prevProps.location !== this.props.location) {
+      this.getHourlyData();
+    }
+  }
+
+  render() {
+    const { data, loading, error } = this.state;
+    const { tempUnit } = this.props;
+    return (
+      <Card>
+        <div id="hourly">
+          <Title title="Hourly forecast" />
+          {data.forecast.length > 0 ? (
+            <>
+              <div
+                className={`${styles.container} ${
+                  this.context.theme === Theme.DARK && styles[Theme.DARK]
+                }`}
+              >
+                <SvgBackground />
+                {data.forecast.map(
+                  ({ time, symbol, temperature, precipAccum }, index, forecastData) => (
+                    <div className={styles.hourContainer} key={time}>
+                      <p className={styles.time}>{time.slice(11, 16)}</p>
+                      <img
+                        className={styles.symbol}
+                        src={ENDPOINTS.symbol + symbol + '.png'}
+                        style={{ top: `${350 - temperature * 10}px` }}
+                      />
+                      <svg height={'400'} width={'40'} className={styles.svgTemperature}>
+                        <CreateLine
+                          coordsBeg={{
+                            x1: '4',
+                            y1: `${400 - temperature * 10}`
+                          }}
+                          coordsEnd={{
+                            x2: `${forecastData[index + 1] ? '40' : '0'}`,
+                            y2: `${
+                              400 - forecastData[index + 1]?.temperature * 10 ||
+                              400 - temperature * 10
+                            }`
+                          }}
+                          color={'red'}
+                        />
+                        <circle
+                          cy={400 - temperature * 10}
+                          cx={'4'}
+                          r="4"
+                          fill={'red'}
+                          className={styles.circle}
+                        ></circle>
+                        <text className={styles.textTemperature} x={'4'} y={385 - temperature * 10}>
```

Comment:
why not replaced with CreateTempScale component?

Reply:
replaced